src/utils/workspaceUtils.ts

```diff
@@ -67,6 +69,38 @@ export class WorkspaceUtils {
             return resolve(staticResourcesPath);
         });
     }
+
+    static getPackageJson(): object {
+        return JSON.parse(
+            fs.readFileSync(
+                path.join(this.getWorkspaceDir(), PACKAGE_JSON),
+                'utf8'
+            )
+        );
+    }
+
+    static setPackageJson(packageJson: object) {
+        fs.writeFileSync(
+            path.join(this.getWorkspaceDir(), PACKAGE_JSON),
+            JSON.stringify(packageJson, null, TAB_SPACES)
+        );
+    }
+
+    static packageJsonExists(): boolean {
+        return fs.existsSync(path.join(this.getWorkspaceDir(), PACKAGE_JSON));
```

Comment:
See my comment below for `isSfdxProjectOpened`.... I think `packageJsonExists()` should never throw and instead should always return either `true` or `false` Since  `getWorkspaceDir()` can throw you should wrap this code in `try-catch` to ensure you always return either `true` or `false`

Reply:
Does `fs.existsSync()` throw error? The return value is boolean so if an error is thrown internally in that method it should be caught and return `false`. So basically it should be what you suggested inside it. Let me check, that's what I thought I read it somewhere.

Reply:
https://github.com/nodejs/node/blob/40565e90b4fc72649cd4801b01d6e0e00f21b06c/lib/fs.js#L283

Reply:
Nvm. You meant getWorkspaceDir can throw. I'm going to fix it. Thanks.